components/ai_chat/resources/page/components/smart_mode_modal/smart_mode_modal.tsx

```diff
@@ -0,0 +1,347 @@
+
+  const promptError = React.useMemo(() => {
+    // Undefined means the user hasn't typed anything yet,
+    // so we don't show an immediate error.
+    if (prompt === undefined) {
+      return ''
+    }
+    if (!prompt.trim()) {
+      return getLocale(S.CHAT_UI_PROMPT_REQUIRED_ERROR)
+    }
+    if (prompt.length > 1000) {
+      return getLocale(S.CHAT_UI_PROMPT_TOO_LONG_ERROR)
+    }
+    return ''
+  }, [prompt])
+
+  // Determine if we're in edit mode
+  const isEditMode = React.useMemo(
+    () => !!aiChatContext.smartModeDialog?.id,
+    [aiChatContext.smartModeDialog?.id],
+  )
+
+  const dialogTitle = React.useMemo(() => {
+    if (showDelete) {
+      return getLocale(S.CHAT_UI_DELETE_SMART_MODE_TITLE)
+    }
+    if (isEditMode) {
+      return getLocale(S.CHAT_UI_EDIT_SMART_MODE_TITLE)
+    }
+    return getLocale(S.CHAT_UI_NEW_SMART_MODE_TITLE)
+  }, [isEditMode, showDelete])
+
+  // Methods
+  const closeAndReset = React.useCallback(() => {
+    aiChatContext.setSmartModeDialog(null)
+    setShortcut(undefined)
+    setPrompt(undefined)
+    setSelectedModel(undefined)
+    setShowDelete(false)
+  }, [aiChatContext.setSmartModeDialog])
+
+  const onSave = React.useCallback(() => {
+    if (
+      shortcutError
+      || promptError
+      || !shortcut
+      || !prompt
+      || !selectedModel
+    ) {
+      return
+    }
+
+    if (isEditMode && aiChatContext.smartModeDialog?.id) {
+      aiChatContext.service?.updateSmartMode(
+        aiChatContext.smartModeDialog.id,
+        shortcut.trim(),
+        prompt.trim(),
+        selectedModel || null,
+      )
+    } else {
+      aiChatContext.service?.createSmartMode(
+        shortcut.trim(),
+        prompt.trim(),
+        selectedModel || null,
+      )
+    }
+
+    closeAndReset()
+  }, [
+    shortcutError,
+    promptError,
+    isEditMode,
+    aiChatContext.smartModeDialog?.id,
+    shortcut,
+    prompt,
+    selectedModel,
+    aiChatContext.service,
+    closeAndReset,
+  ])
+
+  const onDelete = React.useCallback(() => {
+    if (!isEditMode || !aiChatContext.smartModeDialog?.id) {
+      return
+    }
+
+    if (aiChatContext.service?.deleteSmartMode) {
+      aiChatContext.service.deleteSmartMode(aiChatContext.smartModeDialog.id)
+    }
+    closeAndReset()
+  }, [
+    isEditMode,
+    aiChatContext.smartModeDialog?.id,
+    aiChatContext.service,
+    closeAndReset,
+  ])
+
+  return (
+    <Dialog
+      isOpen={!!aiChatContext.smartModeDialog}
+      showClose
+      backdropClickCloses={false}
+      onClose={closeAndReset}
+    >
+      <div slot='title'>{dialogTitle}</div>
+
+      <div className={styles.description}>
+        {getLocale(
+          showDelete
+            ? S.CHAT_UI_DELETE_SMART_MODE_WARNING
+            : S.CHAT_UI_SMART_MODE_DESCRIPTION,
+        )}
+      </div>
+
+      {!showDelete && (
+        <div className={styles.formSection}>
+          <Dropdown
```

Comment:
Hey, this component looks suspiciously like our other model selectors - I wonder if we could share them?